### ISSUE TITLE
Custom expando support

### ISSUE THREAD

**Issue by OctoNezd** [OctoNezd/oldlander#15]
Probably copy RES code for that? Maybe even add RES as git submodule and import features from it?

**Comment by faraplay** [comment]
To be more clear, what features are you looking for in the custom expando?

**Comment by OctoNezd** [comment]
Issue with RES is that it requires access for domains (e.g. twitter, v.redd.it, etc) - which mobile Firefox cant ask for. At first I thought of basically faking RES environment for their scripts to work but it sounds like a bad idea now that I think of it - custom implementation for them would probably be best way around.

**Comment by faraplay** [comment]
A custom implementation sounds like a good idea.
If we're implementing the features ourselves though, it would probably be best to create separate issues for each of the features we want (e.g. ability to zoom, rotate).

**Comment by OctoNezd** [comment]
I think it would be nicer to grab a pre-existing library for that rather than reinventing the wheel, e.g. https://www.lightgalleryjs.com/ (first google search)

**Comment by OctoNezd** [comment]
![image](https://github.com/OctoNezd/oldlander/assets/15308404/88a144c0-a1b9-45aa-81ee-50b6d7e15a54)

lightgallery plays nicely it seems - gonna integrate it

**Comment by faraplay** [comment]
Careful with the license issue, if we want to use lightgallery we need to change our license to GPLv3

**Comment by OctoNezd** [comment]
It seems to me that it is required to use GPLv3 compliant license, no?

**Comment by OctoNezd** [comment]
quote per https://www.lightgalleryjs.com/docs/license/

```Open source license

If you are creating an open source application under a license compatible with the GNU GPL license v3, you may use this project under the terms of the GPLv3. Questions? Read the GPL FAQ.```

If I understand correctly - MIT is compliant. Of course, no one would be able to repackage OldLander under proprietary license after that but I dont think I will sell it to some giant data harvesting company which would make extension closed source.

**Comment by OctoNezd** [comment]
Basic support added in 4ebe41a.

**Comment by OctoNezd** [comment]
v.redd.it (and probably others) can be implemented using video.js, and lightGallery is compatible with it,

**Comment by faraplay** [comment]
> If I understand correctly - MIT is compliant. Of course, no one would be able to repackage OldLander under proprietary license after that but I dont think I will sell it to some giant data harvesting company which would make extension closed source.

I think the compatibility only goes one way (can use MIT license code in GPLv3 project, but can't use GPLv3 code in MIT license project). That's the impression I got from this: https://opensource.stackexchange.com/questions/1640/if-im-using-a-gpl-3-library-in-my-project-can-i-license-my-project-under-mit-l
I assume you want to keep this project open source anyway so there shouldn't be issues in changing to GPLv3? I'm not an expert though.

**Comment by OctoNezd** [comment]
and looking at this - https://tech.popdata.org/the-gpl-license-and-linking-still-unclear-after-30-years/ noone understands this issue.

**Comment by faraplay** [comment]
Alright let's ignore the potential license problem for now then 😅

**Comment by faraplay** [comment]
Dumb question but how do you close the image preview?

**Comment by OctoNezd** [comment]
There is a cross in top left corner and you can click on black background?

**Comment by faraplay** [comment]
Thanks, clicking on the background works. But for me, the top left corner has the image count (e.g. 1/3, although it's usually 1/1) instead of a cross.

**Comment by OctoNezd** [comment]
Sorry, I am stupid and still confuse sides. Right side - see the screenshot I sent before.

**Comment by faraplay** [comment]
There's no cross for me on the right side either, only a zoom-in button. See screenshot:
<img src="https://github.com/OctoNezd/oldlander/assets/65869253/2716f0f4-a88d-4a93-beb5-fd06567469b8" width=360>

**Comment by OctoNezd** [comment]
Hm, thats weird. Can you check whether the button gets created at all?

**Comment by faraplay** [comment]
![image](https://github.com/OctoNezd/oldlander/assets/65869253/2aac4591-e0e5-437f-a572-aa51c1b5bacd)
It looks like there is no close button at all

**Comment by faraplay** [comment]
Now the close button is appearing all of a sudden. I have no idea why it started working now

**Comment by OctoNezd** [comment]
Perhaps lightGallery is not as good as I hoped. Gonna look for alternative libraries later on I guess.

**Comment by faraplay** [comment]
Ok I found out what is making the close button appear/disappear.
Steps to reproduce (on Firefox desktop):

1. Turn on Responsive Design Mode (Ctrl+Shift+M)
2. In the User Agent bar (labelled UA), type in anything with the word "Android". ("iPad", "iPhone", "Windows Phone" also work.)
3. Load up reddit (or refresh if you already have it loaded)
4. Open a lightscreen image preview.
5. The download image and close buttons are now gone.

There might be some setting with user agents/detecting mobile devices in the library you could adjust to fix this?

**Comment by faraplay** [comment]
> Ok I found out what is making the close button appear/disappear. Steps to reproduce (on Firefox desktop):
> ...

now made into separate issue #54

**Comment by OctoNezd** [comment]
vReddIt expandos are blocked by https://github.com/sachinchoolur/lightGallery/issues/1578

**Comment by OctoNezd** [comment]
Nevermind, I am retarded. Seems to be an issue with the way extension loads in.

**Comment by OctoNezd** [comment]
Nevermind, issue seems to be on oldlander side. Perhaps videojs doesn't get injected correctly?

**Comment by OctoNezd** [comment]
It seems that we can just force load dash.js into page, and after that any <video> tag would support application/dash+xml, as far as I understand that.
